X-From-Line: walker@sky.nlm.nih.gov Tue Dec 29 03:12:38 1998
Return-Path: <walker@sky.nlm.nih.gov>
Received: from ncbi.nlm.nih.gov (ncbi.nlm.nih.gov [130.14.25.1]) by mail1.sol.no (8.9.1a/8.9.1/1.12)
	with ESMTP id <DAA00779> for <gisle@aas.no>; Tue, 29 Dec 1998 03:12:37 +0100 (MET)
Received: from sky.nlm.nih.gov (sky [130.14.25.194])
	by ncbi.nlm.nih.gov (8.8.8/8.8.8) with ESMTP id VAA07108;
	Mon, 28 Dec 1998 21:12:35 -0500 (EST)
Received: (from walker@localhost)
	by sky.nlm.nih.gov (8.8.8+Sun/8.8.8) id VAA27503;
	Mon, 28 Dec 1998 21:12:34 -0500 (EST)
Date: Mon, 28 Dec 1998 21:12:34 -0500
From: D Roland Walker <walker@ncbi.nlm.nih.gov>
To: Gisle Aas <gisle@aas.no>
Cc: scotth@sgi.com
Subject: compiling Digest::MD5 under IRIX cc -n32
Message-ID: <19981228211234.A27296@sky.ncbi.nlm.nih.gov>
Mime-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Lines: 71
Xref: furu.g.aas.no box:12442

The default config for perl on IRIX 6.x is to use the

  cc -n32

compiler (if available) with optimization

  -O3

This combination freezes up somewhere in the optimization of

  MD5Transform

in MD5.c from the Digest::MD5 module -- I haven't been able to track down 
exactly where/how.  The following conservative patch solves the problem by
reducing optimizations under IRIX if the n32 or 64 ABI is being used.  The
only version of the IRIX compiler I have handy to test is the Mongoose v7.1;
I don't know how other versions do.

R

# make MD5.c compile under IRIX cc -n32/64
# 
# To apply this patch:
# STEP 1: Chdir to the source directory.
# STEP 2: Run the 'applypatch' program with this patch file as input.
#
# If you do not have 'applypatch', it is part of the 'makepatch' package
# that you can fetch from the Comprehensive Perl Archive Network:
# http://www.perl.com/CPAN/authors/Johan_Vromans/makepatch-x.y.tar.gz
# In the above URL, 'x' should be 2 or higher.
#
# To apply this patch without the use of 'applypatch':
# STEP 1: Chdir to the source directory.
# STEP 2: Run the 'patch' program with this file as input.
#
#### End of Preamble ####

#### Patch data follows ####
diff -c 'Digest-MD5-2.01/Makefile.PL' '/usr/people7/walker/tmp/Digest-MD5-2.01/Makefile.PL'
Index: ./Makefile.PL
*** ./Makefile.PL	Fri Dec 18 10:11:25 1998
--- ./Makefile.PL	Mon Dec 28 20:36:45 1998
***************
*** 5,10 ****
--- 5,11 ----
  
  my @extra;
  @extra = (DEFINE => "-DU32_ALIGNMENT_REQUIRED") unless free_u32_alignment();
+ push @extra, (OPTIMIZE => "-O1") if $^O =~ /irix/ and $Config{cc} =~ /64|n32/;
  
  WriteMakefile(
      'NAME'	   => 'Digest::MD5',
#### End of Patch data ####

#### ApplyPatch data follows ####
# Data version        : 1.0
# Date generated      : Mon Dec 28 20:58:11 1998
# Generated by        : makepatch 2.00
# Recurse directories : Yes
# p 'Makefile.PL' 2460 914895405 0100600
#### End of ApplyPatch data ####

#### End of Patch kit [created: Mon Dec 28 20:58:11 1998] ####
#### Checksum: 43 1471 45064 ####








